framework/stats/stats_common.h

```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2018-2020, Arm Limited and Contributors
+ * Copyright (c) 2020, Broadcom Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <chrono>
+#include <string>
+
+namespace vkb
+{
+/**
+ * @brief Handles of stats to be optionally enabled in @ref Stats
+ */
+enum class StatIndex
+{
+	frame_times,
+	cpu_cycles,
+	cpu_instructions,
+	cpu_cache_miss_ratio,
+	cpu_branch_miss_ratio,
+	gpu_cycles,
+	gpu_vertex_cycles,
+	gpu_tiles,
+	gpu_killed_tiles,
+	gpu_fragment_jobs,
+	gpu_fragment_cycles,
+	gpu_ext_reads,
+	gpu_ext_writes,
+	gpu_ext_read_stalls,
+	gpu_ext_write_stalls,
+	gpu_ext_read_bytes,
+	gpu_ext_write_bytes,
+	gpu_tex_cycles,
+};
+
+struct StatIndexHash
+{
+	template <typename T>
+	std::size_t operator()(T t) const
+	{
+		return static_cast<std::size_t>(t);
+	}
+};
+
+enum class StatScaling
+{
+	// The stat is not scaled
+	None,
+
+	// The stat is scaled by delta time, useful for per-second values
+	ByDeltaTime,
+
+	// The stat is scaled by another counter, useful for ratios
+	ByCounter
+};
+
+enum class CounterSamplingMode
+{
+	/// Sample counters only when calling update()
+	Polling,
+	/// Sample counters continuously, update circular buffers when calling update()
+	Continuous
+};
+
+struct CounterSamplingConfig
+{
+	/// Sampling mode (polling or continuous)
+	CounterSamplingMode mode;
+
+	/// Sampling interval in continuous mode
+	std::chrono::milliseconds interval{1};
+
+	/// Speed of circular buffer updates in continuous mode;
+	/// at speed = 1.0f a new sample is displayed over 1 second.
+	float speed{0.5f};
+};
+
+// Per-statistic graph data
+class StatGraphData
+{
+  public:
+	/**
+		 * @brief Constructs data for the graph
```

Comment:
extra indentation

Reply:
I expected clang-format to take care of that. It seems it doesn't though.
Done